commands/polls/polls.js

```diff
@@ -0,0 +1,119 @@
+                    type: 'string',
+                    validate: options => {
+                        var optionsList = options.split(",");
+                        if (optionsList.length > 1) return true;
+                        return 'Polling options must be greater than one.';
+                    }  
+                },
+                {
+                    key: 'time',
+                    prompt: 'How long should the poll last in minutes?',
+                    type: 'integer',
+                    default: 0,
+                    validate: time => {
+                        if (time >= 0 && time <= 60) return true;
+                        return 'Polling time must be between 0 and 60.';
+                    } 
+                },
+            ]
+        });
+    }
+    
+    run(msg, { question, options, time }) {
+        var emojiList = ['1⃣','2⃣','3⃣','4⃣','5⃣','6⃣','7⃣','8⃣','9⃣','🔟'];
+        var optionsList = options.split(",");
+        
+        var optionsText = "";
+        for (var i = 0; i < optionsList.length; i++) { 
+            optionsText += emojiList[i] + " " + optionsList[i] + "\n";
+        }
+        
+        var embed = new RichEmbed()
+            .setTitle(question)
+            .setDescription(optionsText)
+            .setAuthor(msg.author.username, msg.author.displayAvatarURL)
+            .setColor(0xD53C55) // Green: 0x00AE86
+            .setTimestamp();
+            
+        if (time) {
+            embed.setFooter(`The poll has started and will last ${time} minute(s)`);
+        } else {
+            embed.setFooter(`The poll has started and has no end time`);
+        }
+            
+        //msg.delete(); // Remove the user's command message
+
+        msg.channel.send({embed}) // Definitely use a 2d array here..
+            .then(async function (message) {
+                var reactionArray = [];
+                for (var i = 0; i < optionsList.length; i++) { 
+                    reactionArray[i] = await message.react(emojiList[i]);
+                }
+                
+                if (time) {
+                    setTimeout(() => {
+                        // Re-fetch the message and get reaction counts
+                        message.channel.fetchMessage(message.id)
+                            .then(async function (message) {
+                                var reactionCountsArray = [];
+                                for (var i = 0; i < optionsList.length; i++) {
+                                    reactionCountsArray[i] = message.reactions.get(emojiList[i]).count-1;
+                                }
+
+                                // Find winner(s)
+                                var max = -Infinity, indexMax = [];
+                                for(var i = 0; i < reactionCountsArray.length; ++i)
+                                    if(reactionCountsArray[i] > max) max = reactionCountsArray[i], indexMax = [i];
```

Comment:
I dont know JavaScript that well but I don't think this is correct syntax, even if it is correct its very unclear. Please use a code block for multiple operations, or potentially always use them with IF statements. We will formalise a style guide at some point.

Reply:
ok ill see what i can do

Reply:
fixed in vote.js, fixing it in polls.js rn

Reply:
done